code/parse/sexp_container.cpp

```diff
@@ -329,6 +337,233 @@ sexp_container *get_sexp_container_special(const SCP_string& text, size_t start_
+		}
+
+		const auto modifier = get_list_modifier(text.c_str() + lookHere, true);
+
+		if (modifier == ListModifier::INVALID) {
+			Warning(LOCATION,
+				"sexp_container_replace_refs_with_values() found a list container called %s to replace but the modifer "
+				"is not recognised",
+				container.container_name.c_str());
+			return false;
+		}
+
+		const size_t modifier_length = strlen(get_list_modifier_name(modifier));
+		int data_index = 0;
+		size_t number_length = 0;
+		SCP_string number_string;
+		auto &list_data = container.list_data;
+		auto list_it = list_data.begin();
+
+		switch (modifier) {
+		case ListModifier::GET_FIRST:
+			replacement_text = list_data.front();
+			break;
+
+		case ListModifier::GET_LAST:
+			replacement_text = list_data.back();
+			break;
+
+		case ListModifier::REMOVE_FIRST:
+			replacement_text = list_data.front();
+			list_data.pop_front();
+			break;
+
+		case ListModifier::REMOVE_LAST:
+			replacement_text = list_data.back();
+			list_data.pop_back();
+			break;
+
+		case ListModifier::GET_RANDOM:
+			data_index = util::Random::next((int)list_data.size());
+			replacement_text = *std::next(list_it, data_index);
+			break;
+
+		case ListModifier::REMOVE_RANDOM:
+			data_index = util::Random::next((int)list_data.size());
+			std::advance(list_it, data_index);
+			replacement_text = *list_it;
+			list_data.erase(list_it);
+			break;
+
+		case ListModifier::AT_INDEX:
+			number_string = text.substr(lookHere + modifier_length);
+			Assert(!number_string.empty());
+			data_index = atoi(number_string.c_str());
+			if (data_index < 0 || data_index >= (int)list_data.size()) {
+				Warning(LOCATION,
+					"sexp_container_replace_refs_with_values() found invalid index '%s' into list container %s",
+					number_string.c_str(),
+					container.container_name.c_str());
+				return false;
+			}
+			replacement_text = *std::next(list_it, data_index);
+
+			// include spaces, in case the FREDder put one between "At" and the index
+			number_length = strspn(number_string.c_str(), " 0123456789");
+			break;
+
+		default:
+			Warning(LOCATION, "sexp_container_replace_refs_with_values() found a container called %s to replace but the modifer is not recognised", container.container_name.c_str());
+			return false;
+		}
+
+		num_chars_to_replace += modifier_length;
+
+		// for the "At" modifier, we need to include the chars used by the index
+		if (modifier == ListModifier::AT_INDEX) {
+			num_chars_to_replace += number_length;
+		}
+	}
+
+	// account for the final '&'
+	++num_chars_to_replace;
+
+	return true;
+}
+
+// inspired by sexp_replace_variable_names_with_values()
+/**
+* Replace container references in a string with their values
+**/
+bool sexp_container_replace_refs_with_values(SCP_string &text)
+{
+	bool replaced_anything = false;
+	size_t lookHere = 0;
+	int num_replacements = 0;
+
+	while (lookHere < text.length()) {
+		// look for the meta-character
+		size_t foundHere = text.find(sexp_container::DELIM, lookHere);
```

Comment:
Ah, I was wrong another camel case.  Again, not a huge deal, and I wouldn't hold back approval for it.

Reply:
See my other comment explaining why camel case is used here.